packages/app/ui/components/PostContent/contentUtils.tsx

```diff
@@ -277,6 +278,95 @@ function convertTopLevelInline(verse: ub.VerseInline): BlockData[] {
   return blocks;
 }
 
+// Process inlines to extract embeds as separate blocks
+function extractEmbedsFromInlines(inlines: ub.Inline[]): BlockData[] {
+  const blocks: BlockData[] = [];
+  let currentSegment: ub.Inline[] = [];
+
+  function flushSegment() {
+    if (currentSegment.length > 0) {
+      // Check if segment only contains whitespace
+      const isOnlyWhitespace = currentSegment.every(
+        item => typeof item === 'string' && item.trim() === ''
+      );
+      
+      // Only create a paragraph if there's actual content
+      if (!isOnlyWhitespace) {
+        const convertedInlines = convertInlineContent(currentSegment);
+        
+        // Filter out empty text nodes or text nodes with only whitespace
+        const filteredInlines = convertedInlines.filter(
+          inline => !(inline.type === 'text' && inline.text.trim() === '')
+        );
+        
+        if (filteredInlines.length) {
+          blocks.push({
+            type: 'paragraph',
+            content: filteredInlines,
+          });
+        }
+      }
+      currentSegment = [];
+    }
+  }
+
+  for (const inline of inlines) {
+    // Check if this is a link that matches any of our trusted providers
+    if (ub.isLink(inline)) {
+      const isTrustedEmbed = trustedProviders.some(provider => 
+        provider.regex.test(inline.link.href)
+      );
+      
+      if (isTrustedEmbed) {
+        // Flush the current segment before adding the embed
+        flushSegment();
+
+        // Add the link as a dedicated embed block
+        blocks.push({
+          type: 'embed',
+          url: inline.link.href,
+          content: inline.link.content || inline.link.href,
+        });
+      } else {
+        // Not a trusted embed provider, add to normal paragraph
+        currentSegment.push(inline);
+      }
+    } else {
+      // Not a link, add to normal paragraph
+      currentSegment.push(inline);
+    }
+  }
+
+  // Flush any remaining inlines as a paragraph
+  flushSegment();
+
+  // If no blocks were created (only had non-link inlines), create a single paragraph if needed
```

Comment:
not sure I understand this because the previous line shoves a block in if we have pending segments and seems to do same logic

Reply:
Yep, you're right. I meant to remove this after adding flushSegment. Fixed in latest commit.